packages/permission-controller/src/rpc-methods/getPermissions.test.ts

```diff
@@ -11,20 +11,18 @@ describe('getPermissions RPC method', () => {
 
     const engine = new JsonRpcEngine();
     engine.push((req, res, next, end) =>
-      // TODO: Replace `any` with type
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      implementation(req as any, res as any, next, end, {
+      // @ts-expect-error Abusing types for testing purposes
```

Comment:
?

Reply:
Yes, the type errors being suppressed with `@ts-expect-error` in this file and others seem like the result of loose/incorrect typing, not intentional destructive testing (see also https://github.com/MetaMask/core/pull/4171/files#r1572630610).

Fixes we could apply for this file:

```diff
diff --git a/packages/permission-controller/src/rpc-methods/getPermissions.test.ts b/packages/permission-controller/src/rpc-methods/getPermissions.test.ts
index 4418a2b1..0f65c6ea 100644
--- a/packages/permission-controller/src/rpc-methods/getPermissions.test.ts
+++ b/packages/permission-controller/src/rpc-methods/getPermissions.test.ts
@@ -1,5 +1,12 @@
 import { JsonRpcEngine } from '@metamask/json-rpc-engine';
+import type {
+  Json,
+  JsonRpcRequest,
+  JsonRpcSuccess,
+  PendingJsonRpcResponse,
+} from '@metamask/utils';
 
+import type { PermissionConstraint } from '../Permission';
 import { getPermissionsHandler } from './getPermissions';
 
 describe('getPermissions RPC method', () => {
@@ -10,11 +17,16 @@ describe('getPermissions RPC method', () => {
     });
 
     const engine = new JsonRpcEngine();
-    engine.push((req, res, next, end) =>
-      // @ts-expect-error Abusing types for testing purposes
-      implementation(req, res, next, end, {
-        getPermissionsForOrigin: mockGetPermissionsForOrigin,
-      }),
+    engine.push(
+      (
+        req: JsonRpcRequest<[]>,
+        res: PendingJsonRpcResponse<PermissionConstraint[]>,
+        next,
+        end,
+      ) =>
+        implementation(req, res, next, end, {
+          getPermissionsForOrigin: mockGetPermissionsForOrigin,
+        }),
     );
 
     const response = await engine.handle({
@@ -22,8 +34,11 @@ describe('getPermissions RPC method', () => {
       id: 1,
       method: 'arbitraryName',
     });
-    // @ts-expect-error Abusing types for testing purposes
-    expect(response.result).toStrictEqual(['a', 'b', 'c']);
+    expect((response as JsonRpcSuccess<Json>).result).toStrictEqual([
+      'a',
+      'b',
+      'c',
+    ]);
     expect(mockGetPermissionsForOrigin).toHaveBeenCalledTimes(1);
   });
 
@@ -34,11 +49,16 @@ describe('getPermissions RPC method', () => {
       .mockImplementationOnce(() => null);
 
     const engine = new JsonRpcEngine();
-    engine.push((req, res, next, end) =>
-      // @ts-expect-error Abusing types for testing purposes
-      implementation(req, res, next, end, {
-        getPermissionsForOrigin: mockGetPermissionsForOrigin,
-      }),
+    engine.push(
+      (
+        req: JsonRpcRequest<[]>,
+        res: PendingJsonRpcResponse<PermissionConstraint[]>,
+        next,
+        end,
+      ) =>
+        implementation(req, res, next, end, {
+          getPermissionsForOrigin: mockGetPermissionsForOrigin,
+        }),
     );
 
     const response = await engine.handle({
@@ -46,8 +66,7 @@ describe('getPermissions RPC method', () => {
       id: 1,
       method: 'arbitraryName',
     });
-    // @ts-expect-error Abusing types for testing purposes
-    expect(response.result).toStrictEqual([]);
+    expect((response as JsonRpcSuccess<Json>).result).toStrictEqual([]);
     expect(mockGetPermissionsForOrigin).toHaveBeenCalledTimes(1);
   });
 });
```

Reply:
This is just replacing the previous `any` types with a `@ts-expect-error` directive. Either way, we are "abusing" our types by intentionally violating them. The reason for this "abuse" is that we are testing runtime errors that can only be produced through invalid TypeScript.